firmware/src/handlers/init.rs

```diff
@@ -618,6 +618,31 @@ pub async fn handle_import_seed(
     display_mnemonic(unverified_config, events, peripherals).await
 }
 
+pub async fn handle_show_mnemonic(
+    wallet: Rc<PortalWallet>,
+    mut events: impl Stream<Item = Event> + Unpin,
+    peripherals: &mut HandlerPeripherals,
+) -> Result<CurrentState, Error> {
+    let page = LoadingPage::new();
+    page.init_display(&mut peripherals.display)?;
+    page.draw_to(&mut peripherals.display)?;
+    peripherals.display.flush()?;
+    peripherals.tsc_enabled.enable();
+
+    let mnemonic =
+        Mnemonic::from_entropy(&wallet.config.secret.mnemonic.bytes).map_err(map_err_config)?;
+    let mnemonic_str = mnemonic.word_iter().collect::<alloc::vec::Vec<_>>();
+    for (chunk_index, words) in mnemonic_str.chunks(2).enumerate() {
```

Comment:
this is duplicated from `display_mnemonic`, here we don't need to save intermediate states. We can however refactor out this function with a boolean to save intermediate states and use in both places to save same code space

Reply:
I think unfortunately we will probably need to store the state because if you have 24 words it's very tricky to go through all of them within the 60s imposed by iOS.

But I think that should be part of a future PR in which we also migrate the `display_mnemonic` to use the new checkpoint system, which has the added benefit that can fully resume the state (including the progress bar) and not just the last page shown.